### ISSUE TITLE
enhance TSM support to specify what to backup and restore

### ISSUE THREAD

**Issue by jsmeix** [rear/rear#823]
Up to ReaR version 1.18 when using TSM
there is no way for the user to explicitly specify
what TSM should save in the backup (during "rear mkbackup") and
what TSM should restore from the backup (during "rear recover")
when the default behaviour is not what he actually needs.

Ideally TSM should support already existing "BACKUP"
options in default.conf as far as applicable, e.g.

<pre>
BACKUP_PROG_EXCLUDE
BACKUP_PROG_INCLUDE
or
EXCLUDE_BACKUP
EXCLUDE_RESTORE
</pre>

For details see https://github.com/rear/rear/issues/821

I set "looking for sponsorship" for this issue because
there is nothing at all what I could do regarding
issues with third-party backup tools (or any other
third-party software) that I do not have on my
systems (i.e. where I cannot test anything).

In short:

This issue needs to be implemented by someone
who uses rear with TSM on his systems.

To anybody who uses ReaR with TSM on his systems:

Just be courageous and enhance the curently rather
poor TSM support in ReaR - it cannot get worse ;-)

**Comment by jsmeix** [comment]
https://github.com/rear/rear/pull/833 mitigates the problem where the btrfs subvlums are
not restored by default via TSM.

Now by default both 'fs' and 'btrfsmountedsubvol' entries
in disklayout.conf are used to generate the default TSM
filespaces to include in a restore.

This mitigates https://github.com/rear/rear/issues/821
because now a better default is used but this does
not yet implement https://github.com/rear/rear/issues/823
because the user still cannot explicitly specify what to backup
and restore with TSM when the default behaviour is not
what he actually needs.

**Comment by gdha** [comment]
@schabrolles Could you have a look at this issue and tell us if it makes sense? If not, then we better close this issue, no?

**Comment by jsmeix** [comment]
@gdha 
it think item (4.) in
https://github.com/rear/rear/pull/1348#issuecomment-356556742
could be related.

**Comment by schabrolles** [comment]
@gdha 

I start an implementation in #1348 
Currently, it only provides a list of Directories or FS to add to the TSM backup. (From `BACKUP_PROG_INCLUDE`).
=> This allow to backup only part of the system (only "systemvg" for example with ONLY_INCLUDE_VG=( "systemvg" )). In that case, only the FS part of SystemVG will be included into the TSM backup.

But, It is more difficult to add an exclude list (there is no option in the dsmc cli)

By default, TSM uses special files (include exclude files) which tell what to backup and what to exclude.
But changing those files from ReaR seems a bit dangerous... I would prefer let the user modifying directly those files.

**Comment by jsmeix** [comment]
@schabrolles 
I fully agree to not overdo the TSM support in ReaR.

When TSM has its own files how to configure TSM backup (and restore)
ReaR should never ever change those TSM config files because
in general all files on the original system are sacrosanct for ReaR.

But in case of BACKUP=TSM
ReaR may tell the user when he has (non-empty)settings in
BACKUP_PROG_EXCLUDE
BACKUP_PROG_INCLUDE
EXCLUDE_BACKUP
EXCLUDE_RESTORE
that are ignored by TSM and that such things
must be set in TSM own config files
or at least this should be described in
the "BACKUP=TSM" section in default.conf

In current default.conf the EXCLUDE_BACKUP description
looks like a globally available config option, cf.
http://lists.relax-and-recover.org/pipermail/rear-users/2016-January/003276.html
which is basically the same kind of issue for NSR.
